src/main/java/io/vertx/ext/mongo/MongoClient.java

```diff
@@ -69,6 +71,18 @@ static MongoClient createShared(Vertx vertx, JsonObject config) {
     return new MongoClientImpl(vertx, config, DEFAULT_POOL_NAME);
   }
 
+  /**
+   * Constructor targeting the jvm
+   * @param vertx the Vert.x instance
+   * @param config the configuration use only to provide objectId and database name
+   * @param dataSourceName the data source name
+   * @param settings the native java mongo settings
+   * @return the client
+   */
+  @GenIgnore
+  static MongoClient createShared(Vertx vertx, JsonObject config, String dataSourceName, MongoClientSettings settings) {
```

Comment:
Probably needs fuller Javadoc and I think `createdShared` isn't quite what this is for?

Reply:
Fix it.